design/one-pager-template-stacks.md

```diff
@@ -0,0 +1,1009 @@
+  names:
+    plural: helloworlds
+    singular: helloworld
+    kind: HelloWorld
+```
+
+</details>
+
+### Plus One
+
+The following demonstrates a `PlusOne` resource `Stack` which would append "+ "
+to the single `status` field of the resource on each reconciliation.
+
+```yaml
+apiVersion: stacks.crossplane.io/v1alpha1
+kind: Stack
+metadata:
+  name: PlusOne
+spec:
+  customresourcedefinitions:
+  - kind: PlusOne
+    apiVersion: plusses.crossplane.example.com/v1
+  templateStatus:
+    plusses.crossplane.example.com/v1: |
+      output: "+ {{.status.output}}"
+```
+
+```yaml
+apiVersion: plusses.crossplane.example.com
+kind: PlusOne
+metadata:
+  name: plusses
+```
+
+After the first reconciliation, this resource will have a `status.output` value
+of `+ `.  On the second pass, `+ + `, third, `+ + + `, and so on until the size
+of this resource can no longer be accommodated.
+
+Basic string concatenation has not been demonstrated because the TSM template
+engine may or may not provide math functions like addition.
+
+<details>
+<summary>CRD for PlusOne</summary>
+
+```yaml
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  name: plusone.crossplane.example.com
+spec:
+  group: crossplane.example.com
+  versions:
+    - name: v1
+      served: true
+      storage: true
+      schema:
+        openAPIV3Schema:
+          type: object
+          properties:
+            status:
+              type: object
+              properties:
+                output:
+                  type: string
+      subresources:
+        status: {}
+  scope: Namespaced
+  names:
+    plural: plusones
+    singular: plusone
+    kind: PlusOne
+```
+
+</details>
+
+### Usage Walk-through
+
+If a developer has a CRD for `foo.group/version` and they want to use the Stack Manager and Template Stacks to handle these, they could create a Template Stack containing the relevant templates and other Stack metadata (out of scope here), and then install it:
+
+```sh
+# this syntax is just an example
+kubectl crossplane stack install example/foo-template-stack
+```
+
+This creates a `StackInstall` resource:
+
+```yaml
+apiVersion: stacks.crossplane.io/v1alpha1
+kind: StackInstall
+metadata:
+  name: Foo
+spec:
+  package: example/foo-template-stack:latest
+```
+
+In response to a `StackInstall` request, the Stack Manager will do [its normal
+work](https://github.com/crossplaneio/crossplane/blob/master/design/design-doc-stacks.md#installation-flow)
+and [some extra work](#Resource-life-cycle) to create a `Stack` resource and
+apply the appropriate `template` and `templateStatus` bodies to that `Stack`
+resource. It will also create a `Deployment` and `ServiceAccount` for the TSM.
+
+```yaml
+apiVersion: stacks.crossplane.io/v1alpha1
+kind: Stack
+metadata:
+  name: Foo
+spec:
+  customresourcedefinitions:
+  - kind: Foo
+    apiVersion: group/version
+  templates:
+    # the GVK for the CRD the Stack consumer will create
+    foo.group/version:
```

Comment:
This should be updated to include `sources`, `behaviors`, `resources`, `defaults`, and `status` (per #956).

https://github.com/crossplaneio/crossplane/pull/956/files#diff-e81d9b5b6cbb8b4bdfc4d7a5ef0f2f25R261-R293

Reply:
Deferring these updates to #956.